Assets/Scripts/Hurt.cs

```diff
@@ -0,0 +1,58 @@
+using UnityEngine;
+using UnityEngine.Events;
+
+// Interface for objects that can be hurt
+public interface IHurtable
+{
+    void TakeDamage(float amount, GameObject damageSource);
+    void SetInvulnerable(bool invulnerable);
+    bool IsInvulnerable();
+}
+
+public class Hurt : MonoBehaviour, IHurtable
+{
+    [Header("Hurt Settings")]
+    [SerializeField] private bool isInvulnerable = false;
+    [SerializeField] private bool debugMode = false;
+
+    [Header("Events")]
+    public UnityEvent<float, GameObject> OnDamaged;
```

Comment:
Interesting, I've never used the `UnityEvent` type before and always default to `Action` which is the underlying C# event. Do you know what the differences are, if any?